### PR TITLE
[OneLoc] Localize Microsoft.Android.Templates

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://github.com/xamarin/xamarin-android/pull/6677

The `Microsoft.Android.Templates` project has been onboarded to the new
template localization system.  Local template builds will now use
`Microsoft.TemplateEngine.Tasks` to generate `templatestrings.*.json`
files used for localization.

The new `update-locproject.ps1` script runs as part of the `OneLocBuild`
job in CI.  This script will copy each english `templatestrings.en.json`
file to `templatestrings.json`, and add it as a new translation source
file entry to `LocProject.json`.  This update to `LocProject.json` will
tell the `OneLocBuild` task to process our templates.

This script borrows heavily from the [generate-locproject.ps1][0] script
found in dotnet/arcade.

-----

Description from PR 6677:

Template engine (dotnet new) added support for template localizations in
.NET 6.0. This new system replaces the old way of localizing templates
that only worked on Visual Studio and works on both .NET CLI as well as
Visual Studio.

This PR introduces changes to switch to the new template localization system.

Summary of the changes:

 * Adds `<LocalizeTemplates>true</>` to project containing templates.
 * Include PackageReference for `Microsoft.TemplateEngine.Tasks` to project containing templates.
 * Generates localization files to be translated by loc team

Every time there is a change to one of the templates:

 * The dev making the change should build the modified template project. This will update the loc files on the local working copy,
 * Push the loc files together with the template modifications. Review & merge.
 * OneLocBuild integration will automatically pick up the changes and will send them for translation.
 * You will receive a PR containing the translated template loc files when they are ready. Review & merge.

[0]: https://github.com/dotnet/arcade/blob/e771d68edea1cfcd4a49e160093a4f32df8a6288/eng/common/generate-locproject.ps1

### CODE REVIEW

**Comment by pjcollins** [comment]
Tested this with https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6540194&view=logs&j=b0d940c4-07db-5357-83ba-1a4207a17644&t=f6fbc6c3-6cd9-5e85-c68a-d9367c5fa9a7&l=170. The updated LocProject and OneLocBuild task output look good as far as I can tell, we'll now need to wait for translations for these:

    OneLocBuildClient.exe Information: 0 : [src\Microsoft.Android.Templates\android\.template.config\localize\templatestrings.json] 0% localized for 13 languages.
    OneLocBuildClient.exe Information: 0 : [src\Microsoft.Android.Templates\android-activity\.template.config\localize\templatestrings.json] 0% localized for 13 languages.
    OneLocBuildClient.exe Information: 0 : [src\Microsoft.Android.Templates\android-bindinglib\.template.config\localize\templatestrings.json] 0% localized for 13 languages.
    OneLocBuildClient.exe Information: 0 : [src\Microsoft.Android.Templates\android-layout\.template.config\localize\templatestrings.json] 0% localized for 13 languages.
    OneLocBuildClient.exe Information: 0 : [src\Microsoft.Android.Templates\android-wear\.template.config\localize\templatestrings.json] 0% localized for 13 languages.
    OneLocBuildClient.exe Information: 0 : [src\Microsoft.Android.Templates\androidlib\.template.config\localize\templatestrings.json] 0% localized for 13 languages.

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -16,6 +16,10 @@
       <Uri>https://github.com/dotnet/emsdk</Uri>
       <Sha>7b2cd1ee7169143248a7da9fd749caf22affa624</Sha>
     </Dependency>
+    <Dependency Name="Microsoft.TemplateEngine.Tasks" Version="7.0.100-rc.1.22409.14">
+      <Uri>https://github.com/dotnet/templating</Uri>
+      <Sha>efd5e59cbe2751b14f5331d122ccba27dd788cf4</Sha>
+    </Dependency>
```
Comment (jonathanpeppers) [inline eng/Version.Details.xml:22]: Should this go under `ToolsetDependencies` or `ProductDependencies`? I think it mainly just shows something different on Maestro's website depending on what you pick.

Comment (pjcollins) [reply eng/Version.Details.xml:22]: I'm not sure, [maui](https://github.com/dotnet/maui/blob/8e8b2973c7ca42082957e9b60d9407c85d158608/eng/Version.Details.xml#L70) uses product but [macios](https://github.com/xamarin/xamarin-macios/blob/4221018abada0941f34a787fc4d40561e296db3e/eng/Version.Details.xml#L30) uses toolset. I think toolset probably makes more sense for how it is used so I will move it.

**Review by jonathanpeppers — APPROVED** [review]

File: Localize/update-locproject.ps1 [hunk 2]
```diff
@@ -0,0 +1,28 @@
+param ($SourcesDirectory, $LocProjectPath)
```
Comment (jonpryor) [inline Localize/update-locproject.ps1:1]: 1. Should this use a [shebang](https://stackoverflow.com/a/48218913) and be `+x` on Unix?
 2. If we *do* use a shebang, then this file needs to be Unix formatted.
 3. If this file needs to be Unix formatted, then we should update `.gitattributes` accordingly.

Comment (pjcollins) [reply Localize/update-locproject.ps1:1]: I would lean towards no as I don't anticipate us needing to invoke this script much (if ever) outside of CI, and running the script with a `pwsh` prefix in those cases doesn't feel like too much overhead. I will make these changes if you would prefer though?

**Comment by jonpryor** [comment]
```markdown
Context: https://github.com/dotnet/templating/blob/6ecea73aad1ce3444dbf28fc340e5f464f6278df/docs/Localization.md

The `dotnet new` template engine supports localizing templates as of
.NET 6.

Update the `Microsoft.Android.Templates` project to use the new
localization system.  Local template builds now use
`Microsoft.TemplateEngine.Tasks` teo generate the
`templatestrings.*.json` files used for localization.

The new `update-locproject.ps1` script runs as part of the
`OneLocBuild` job in CI (fa9629db).  This script will copy each
english `templatestrings.en.json` file to `templatestrings.json`, and
add it as a new translation source file entry to `LocProject.json`.
This update to `LocProject.json` will tell the `OneLocBuild` task to
process our templates.

This script borrows heavily from the [generate-locproject.ps1][0]
script found in dotnet/arcade.

[0]: https://github.com/dotnet/arcade/blob/e771d68edea1cfcd4a49e160093a4f32df8a6288/eng/common/generate-locproject.ps1
```

**Pull request merged**